### ISSUE TITLE
Add dataPath for forms and templates

### ISSUE THREAD

**Issue by alexander-schranz** [sulu/sulu#5060]
# Problem description

Currently when I have a custom entity which is extended with templating e.g. using the SuluContentBundle the data need for Developer Experience on the root of the entity e.g.:

```json

{
    "id": "1",
     "title": "My Title"
}
```

and in the xml template it is:

```xml
<?xml version="1.0" ?>
<template>
    <key>default</key>
    <!-- ...  -->

    <properties>
        <property name="title" type="text_line" mandatory="true">
            <meta>
                <title lang="en">Title</title>
                <title lang="de">Titel</title>
            </meta>
        </property>
    <properties>
</template>
```

I personally would prefer that the json can look like this to avoid conflicts with entity data:

```json
{
    "id": "1",
    "templateData": {
         "title": "My Title"
     }
}
```

but the xml should not change he should not be force to write `templateData/title`. 

# Proposed solution

The proposed would be that the FormView can have a `dataPath` parameter.

E.g.:

```php
$view->setOption('dataPath', 'template');
```

**Comment by danrot** [comment]
Personally I would prefer the more explicit approach, and keep it as it is... I have several reasons for that:

1. It is hard to follow, because you have to combine information from two different places (the `View` object in the `Admin` class and the form XML configuration) in order to know where your property is stored, and you have to know about this option (which makes it harder to get into this)
2. This solution does not work anymore, if we want to also use the form for data from a different sub tree
3. I think adding `templateData/` in front of every template is not too much to ask, makes things clearer and avoids the previously mentioned drawbacks.

**Comment by alexander-schranz** [comment]
The developer who create the API and configure the view is mostly the same.

The developer who creates the templates is mostly another and he should be able to use xincludes which are used in pages also on his custom entity.

That there are things we need to check when implementing I'm sure, from DX perspective this would be a nice feature.

**Comment by danrot** [comment]
I think this feature will also have a negative impact on DX, because it is hard to find and if somebody has used it, it might even confuse themselves in the future.
